src/ClusterManager.js

```diff
@@ -349,6 +349,9 @@ class ClusterManager extends EventEmitter {
   * @private
   */
   evalOnCluster(script, options) {
+    if(options.hasOwnProperty('shard')){
+        options.cluster = parseInt(Object.entries(Object.assign({}, this.shardclusterlist)).find(i => i[1].includes(options.shard))?.[0]);
```

Comment:
Instead of `options.cluster = parseInt(Object.entries(Object.assign({}, this.shardclusterlist)).find(i => i[1].includes(options.shard))?.[0]);`

use 
`options.cluster = parseInt(Object.entries(Object.assign({}, this.shardclusterlist)).find(i => i[1].includes(options.shard)) ? i[1].includes(options.shard))[0]) : 0;`